routing/result_interpretation.go

```diff
@@ -0,0 +1,275 @@
+package routing
+
+import (
+	"github.com/lightningnetwork/lnd/channeldb"
+
+	"github.com/lightningnetwork/lnd/lnwire"
+	"github.com/lightningnetwork/lnd/routing/route"
+)
+
+// interpretedResult contains the result of the interpretation of a payment
+// outcome.
+type interpretedResult struct {
+	nodeFailures       map[route.Vertex]struct{}
+	pairResults        map[DirectedNodePair]lnwire.MilliSatoshi
+	finalFailureReason *channeldb.FailureReason
+	policyFailure      *DirectedNodePair
+}
+
+// interpretResult interprets a payment outcome and returns an object that
+// contains information required to update mission control.
+func interpretResult(rt *route.Route, failureSrcIdx *int,
+	failure lnwire.FailureMessage) *interpretedResult {
+
+	i := &interpretedResult{
+		nodeFailures: make(map[route.Vertex]struct{}),
+		pairResults:  make(map[DirectedNodePair]lnwire.MilliSatoshi),
+	}
+
+	final, reason := i.processFail(rt, failureSrcIdx, failure)
+	if final {
+		i.finalFailureReason = &reason
+	}
+
+	return i
+}
+
+// processFail processes a failed payment attempt.
+func (i *interpretedResult) processFail(
+	rt *route.Route, errSourceIdx *int,
+	failure lnwire.FailureMessage) (bool, channeldb.FailureReason) {
```

Comment:
For the same reason it was originally introduced: because we can't take references to the constant returned in this function. But this will be further cleaned up in following commits.